mesa/experimental/cell_space/cell_collection.py

```diff
@@ -0,0 +1,67 @@
+from __future__ import annotations
+
+import itertools
+from collections.abc import Iterable
+from functools import cached_property
+from random import Random
+from typing import TYPE_CHECKING, Callable
+
+if TYPE_CHECKING:
+    from mesa.experimental.cell_space.cell import Cell
+    from mesa.experimental.cell_space.cell_agent import CellAgent
+
+
+class CellCollection:
+    def __init__(
+        self,
+        cells: dict[Cell, list[CellAgent]] | Iterable[Cell],
```

Comment:
If a `CellCollection` can be both an dictionary and iterable of cells, doesn't that cause unexpected behavior? I understand that when iterating though a dict, you iterate though the keys, but now you can't be sure if there are agents in there or not. Mabye we need two seperate constructs.

Reply:
Check the implementation if statement. 

You cannot overload initializers (note:: __init__ is **not** a constructor, __new__ is the constructor).